app/controllers/application_controller/miq_request_methods.rb

```diff
@@ -855,6 +855,10 @@ def prov_set_form_vars(req = nil)
         end
       end
       @edit[:new][:src_vm_id] = [nil, nil] unless @edit[:new][:src_vm_id]
+      # Check that the provisioning template exists
+      if @edit[:wf].request_type == 'template' && @edit[:new][:src_vm_id][0].present?
+        @edit[:new][:src_vm_id][0] = nil if MiqTemplate.find_by(:id => @edit[:new][:src_vm_id][0]).nil?
```

Comment:
`MiqTemplate.find_by(:id => @edit[:new][:src_vm_id][0]).nil?`
vs
`MiqTemplate.exists?(:id => @edit[:new][:src_vm_id][0])` or possibly `MiqTemplate.exist?(@edit[:new][:src_vm_id][0])`

it might make little difference in this particular case but generally `exists?` generates a `LIMIT 1` on the SQL level and the name better matches the intent of what you are doing.

Reply:
Fixed.